src/utils/clientHelpers.ts

```diff
@@ -1,18 +1,55 @@
-import { Executable, InitializeResult } from 'vscode-languageclient/node';
+import * as net from 'net';
+import * as vscode from 'vscode';
+import { Executable, InitializeResult, ServerOptions, StreamInfo } from 'vscode-languageclient/node';
 import { config } from './vscode';
 import { ServerPath } from './serverPath';
+import { outputChannel } from '../extension';
+
+export async function getServerOptions(lsPath: ServerPath): Promise<ServerOptions> {
+  let serverOptions: ServerOptions;
+
+  const port: number | undefined = config('terraform').get('experimentalFeatures.languageServer.tcp.port');
```

Comment:
```suggestion
  const port = config('terraform').get<number>('experimentalFeatures.languageServer.tcp.port');
```

We can pass a type to `get`, so we don't have to hard-code it.

Reply:
This PR has been open so long, I've forgotten parts!

Your suggestions work because of setting `"type" :["number",null ]` and `"default": null` for `terraform.experimentalFeatures.languageServer.tcp.port`. If you remove those, it will default to 0 and cause the behavior I describe.

So, leaving the setting attributes up and above and your suggestions works together.